### PR TITLE
Remove tyrus dependencies in pinot-tools module

### PR DESCRIPTION
**Pull request by jackjlli**
## Description
This PR is to remove the tyrus dependencies, because it contains a class that happens to have the same class name `ServerConfiguration` and same method as the one of `org.glassfish.jersey.containers:jersey-container-grizzly2-http`.

That's why the following exception is thrown when running `RealtimeQuickStart`.
```

Exception in thread "main" java.lang.NoSuchMethodError: org.glassfish.grizzly.http.server.ServerConfiguration.addHttpHandler(Lorg/glassfish/grizzly/http/server/HttpHandler;[Lorg/glassfish/grizzly/http/server/HttpHandlerRegistration;)V
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpServerFactory.createHttpServer(GrizzlyHttpServerFactory.java:258)
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpServerFactory.createHttpServer(GrizzlyHttpServerFactory.java:114)
	at org.apache.pinot.controller.api.ControllerAdminApiApplication.start(ControllerAdminApiApplication.java:115)
	at org.apache.pinot.controller.ControllerStarter.setUpPinotController(ControllerStarter.java:416)
	at org.apache.pinot.controller.ControllerStarter.start(ControllerStarter.java:287)
	at org.apache.pinot.tools.service.PinotServiceManager.startController(PinotServiceManager.java:113)
	at org.apache.pinot.tools.service.PinotServiceManager.startRole(PinotServiceManager.java:90)
	at org.apache.pinot.tools.admin.command.StartServiceManagerCommand.lambda$startBootstrapServices$0(StartServiceManagerCommand.java:234)
	at org.apache.pinot.tools.admin.command.StartServiceManagerCommand.startPinotService(StartServiceManagerCommand.java:286)
	at org.apache.pinot.tools.admin.command.StartServiceManagerCommand.startBootstrapServices(StartServiceManagerCommand.java:233)
	at org.apache.pinot.tools.admin.command.StartServiceManagerCommand.execute(StartServiceManagerCommand.java:183)
	at org.apache.pinot.tools.admin.command.StartControllerCommand.execute(StartControllerCommand.java:130)
	at org.apache.pinot.tools.admin.command.QuickstartRunner.startControllers(QuickstartRunner.java:105)
	at org.apache.pinot.tools.admin.command.QuickstartRunner.startAll(QuickstartRunner.java:140)
	at org.apache.pinot.tools.RealtimeQuickStart.execute(RealtimeQuickStart.java:90)
	at org.apache.pinot.tools.RealtimeQuickStart.main(RealtimeQuickStart.java:46)
```

### CODE REVIEW

File: pinot-tools/src/main/java/org/apache/pinot/tools/streams/MeetupRsvpStream.java [hunk 1]
```diff
@@ -20,102 +20,104 @@
...
   private static final Logger LOGGER = LoggerFactory.getLogger(MeetupRsvpStream.class);
   private StreamDataProducer producer;
   private boolean keepPublishing = true;
-  private ClientManager client;
 
-  public MeetupRsvpStream()
+  private ExecutorService _service;
+  private File _dataFile;
+
+  public MeetupRsvpStream(File dataFile)
```
Comment (mayankshriv) [inline pinot-tools/src/main/java/org/apache/pinot/tools/streams/MeetupRsvpStream.java:46]: I think getting a live stream was there for a reason. Perhaps we should evaluate if that is a good idea, and if so, still keep it, as opposed to having a dataFile. Once it is a data-file, it is the same as any other provider (not necessarily meetup).

**Review by mayankshriv — COMMENTED** [review]
It is unclear on why removing tyrus requires changing MeetupStream to be a static file. Could you explain?

**Comment by kishoreg** [comment]
can you please make sure quickstart (both offline and realtime) works

**Comment by jackjlli** [comment]
> It is unclear on why removing tyrus requires changing MeetupStream to be a static file. Could you explain?

Currently Tyrus is only used for streaming data in `RealtimeQuickStart` class.  It depends on an external source for now (i.e. meetup.com). If anything breaks in the upstream, our build/test will inevitably fail (this is what we faced recently).
Plus, the current logic doesn't validate the actual result from this event because the result isn't deterministic; it just evaluates whether realtime table has consumed some data. Emitting some sample events from a sample file should suffice this requirement.

**Comment by jackjlli** [comment]
> can you please make sure quickstart (both offline and realtime) works

I've tested it in both pinot-tools and pinot-distributions modules and all quick start tests work.

**Comment by mayankshriv** [comment]
> > can you please make sure quickstart (both offline and realtime) works
> 
> I've tested it in both pinot-tools and pinot-distributions modules and all quick start tests work.

Let's also make sure that PerfBenchmarkRunner also works as well.

**Comment by jackjlli** [comment]
> > > can you please make sure quickstart (both offline and realtime) works
> > 
> > 
> > I've tested it in both pinot-tools and pinot-distributions modules and all quick start tests work.
> 
> Let's also make sure that PerfBenchmarkRunner also works as well.

That's been tested as well.

**Comment by Jackie-Jiang** [comment]
I'm against changing the data to a static file. The purpose of the quick start is letting the user play with Pinot and they should be able to see the latest events from a real stream instead of the unchanged records. Is there a way to solve this issue without hardcode the data file?
Also, I'm not able to reproduce this issue locally. Did I miss something?

**Comment by jackjlli** [comment]
@Jackie-Jiang The root cause is that in some machine the `tyrus` jar appears before `jersey-container-grizzly2-http`. 
The reason why this issue cannot be reproduced in every machine is that each OS controls the order of jars listed in the classpath.

**Comment by mayankshriv** [comment]
> I'm against changing the data to a static file. The purpose of the quick start is letting the user play with Pinot and they should be able to see the latest events from a real stream instead of the unchanged records. Is there a way to solve this issue without hardcode the data file?
> Also, I'm not able to reproduce this issue locally. Did I miss something?

@Jackie-Jiang The tyrus dependency causes a lot of issues for me as well, e.g. when running perfBenchmarkRunner locally.
I agree we should have recipe/setup/demo using live stream. However, note that quickStart is also being used as a test, which makes it brittle if upstream has an issue (we had this recently).

My recommendation is to have the test part of quick-start read data from a deterministic source (e.g. local stream or static file). However, in this case, quick-start (as a user demo tool) remains untested. For this, we could have a quick-start test based on real stream only when cutting the release (instead of all commits). What do you think?

**Comment by Jackie-Jiang** [comment]
@mayankshriv @jackjlli I'm okay using static file for testing purpose only, but if we remove `tyrus`, does that mean we can no longer consume real streaming events in the quick-start? I feel that is very bad for demo purpose, or for new users to try out Pinot.

**Comment by Jackie-Jiang** [comment]
Isn't this just a version conflict? Does excluding this conflicting library from one of the library work?

**Comment by jackjlli** [comment]
> Isn't this just a version conflict? Does excluding this conflicting library from one of the library work?

We cannot exclude either one of them because `tyrus` is needed to stream external data and `jersey-container-grizzly2-http` is needed to start the web server.

**Comment by Jackie-Jiang** [comment]
> We cannot exclude either one of them because `tyrus` is needed to stream external data and `jersey-container-grizzly2-http` is needed to start the web server.

You can exclude the underlying dependencies of them to resolve the version conflict right?

**Comment by jackjlli** [comment]
> > We cannot exclude either one of them because `tyrus` is needed to stream external data and `jersey-container-grizzly2-http` is needed to start the web server.
> 
> You can exclude the underlying dependencies of them to resolve the version conflict right?

No, both dependencies are required. We cannot exclude either one of them.

**Comment by Jackie-Jiang** [comment]
@jackjlli I see the problem, basically `tyrus` copied the code from `grizzly` and put it in one package. If you upgrade `tyrus` to `1.15` to be compatible with `grizzly 2.4.4`, then the problem should be solved

**Comment by jackjlli** [comment]
@Jackie-Jiang Thanks for looking into it. I've opened a new PR to bump up the Tyrus version to 1.15 (https://github.com/apache/incubator-pinot/pull/6162). Once that's merged, we can close this PR.

**Comment by jackjlli** [comment]
Closing this PR since https://github.com/apache/incubator-pinot/pull/6162 has been merged.

**Pull request closed without merging**